mkt/extensions/models.py

```diff
@@ -483,19 +483,18 @@ def handle_file_upload_operations(self, upload):
 
         Return the file size."""
         upload.path = smart_path(nfd_str(upload.path))
+        destination = self.file_path
 
         if private_storage.exists(self.file_path):
-            # The filename should not exist. If it does, it means we are trying
-            # to re-upload the same version. This should have been caught
-            # before, so just raise an exception.
-            raise RuntimeError(
-                'Trying to upload a file to a destination that already exists:'
-                ' %s' % self.file_path)
+            destination = private_storage.get_available_name(self.file_path)
```

Comment:
I'm pretty sure that this will work in production, but our mock file storage system doesn't appear to handle this well. @diox, thoughts on the approach?